Realm/RLMQueryUtil.mm

```diff
@@ -85,12 +86,14 @@ size_t find_first(size_t start, size_t end) const override
     }
     void set_table(const Table*) override {}
     const Table* get_table() const override { return nullptr; }
+    std::unique_ptr<Expression> clone(QueryNodeHandoverPatches*) const override { return nullptr; }
```

Comment:
AFAICT from looking at the code that calls this, it does need to return a new object or OrNote::find_first_local() will deref a null pointer.